tensorflow/compiler/mlir/tosa/transforms/legalize_common.cc

```diff
@@ -2577,6 +2566,197 @@ llvm::Optional<Value> convertReduceMeanOp(
   return val;
 }
 
+// Lowers ResizeBilinear and ResizeNearestNeighbor to TOSA resize.
+llvm::Optional<Value> convertResizeOp(PatternRewriter& rewriter, Operation* op,
+                                      RankedTensorType output_type,
+                                      Value input_value, StringRef mode,
+                                      bool align_corners,
+                                      bool half_pixel_centers) {
+  RankedTensorType input_type =
+      input_value.getType().dyn_cast<RankedTensorType>();
+  if (!input_type) return llvm::None;
+
+  auto input_shape = input_type.getShape();
+  auto output_shape = output_type.getShape();
+
```

Comment:
Please verify that the input and output shape are rank 3. A comment as to what the dims are (symbolically) wouldn't hurt. I assume [batch, height, width]?

Reply:
Yes I can add the check in next round.

Isn't NHWC, which is rank 4, being the default format for resize's input/output?